src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,43 @@
-import React from 'react';
+import { TodoCompletedCategory } from '../../types/todoCompletedCategory';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
+
+const nameForCategoriesOption = {
+  [TodoCompletedCategory.all]: 'All',
+  [TodoCompletedCategory.active]: 'Active',
+  [TodoCompletedCategory.competed]: 'Competed',
+};
+
+export const TodoFilter = () => {
+  const { query, category: todoCategory } = useAppSelector(
+    state => state.filter,
+  );
+  const dispatch = useAppDispatch();
+
+  const changeQuery = (newQuery: string) => {
+    dispatch(filterSlice.actions.changeQuery(newQuery));
+  };
+
+  const changeTodoCategory = (newCategory: TodoCompletedCategory) => {
+    dispatch(filterSlice.actions.changeStatus(newCategory));
+  };
 
-export const TodoFilter: React.FC = () => {
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            value={todoCategory}
+            onChange={event =>
+              changeTodoCategory(event.target.value as TodoCompletedCategory)
```

Comment:
Ensure that `event.target.value` is a valid `TodoCompletedCategory`. The current implementation casts it directly, which might lead to runtime errors if the value is not valid. Consider validating the value before casting.